src/electron/process_manager.ts

```diff
@@ -473,6 +549,21 @@ export function teardownVpn() {
   ]);
 }
 
+function getTunDeviceName() {
+  if (isWindows) {
+    tunDeviceName = `tap0901:${TUN2SOCKS_TAP_DEVICE_NAME}:${TUN2SOCKS_TAP_DEVICE_IP}:${
+        TUN2SOCKS_TAP_DEVICE_NETWORK}:${TUN2SOCKS_VIRTUAL_ROUTER_NETMASK}`;
+  } else if (isLinux) {
+    return routingService.getDeviceName()
+        .then((response) => {
+          tunDeviceName = response;
+        })
+        .catch((e) => {
+          console.log(`could not get device name: ${e.message}`);
```

Comment:
Swallowing exceptions is a huge code smell - it's much better to give the client an opportunity to handle the exception.